src/test/java/com/redhat/cloud/notifications/events/EventConsumerTest.java

```diff
@@ -0,0 +1,90 @@
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static com.redhat.cloud.notifications.TestHelpers.serializeAction;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class EventConsumerTest {
+
+    @Inject
+    @Connector("smallrye-in-memory")
+    InMemoryConnector inMemoryConnector;
+
+    @InjectMock
+    EndpointProcessor destinations;
+
+    private Counter rejectedCounter = Metrics.counter("input.rejected");
+    private Counter processingErrorCounter = Metrics.counter("input.processing.error");
+
+    @Test
+    void testValidMessagePayload() throws IOException {
+        Action action = buildValidAction();
+        String serializedAction = serializeAction(action);
+        assertCountersIncrement(0, 0,
+                () -> inMemoryConnector.source("ingress").send(serializedAction));
+        verify(destinations, times(1)).process(eq(action));
+    }
+
+    @Test
+    void testInvalidMessagePayload() {
+        assertCountersIncrement(1, 0,
+                () -> inMemoryConnector.source("ingress").send("I am not a valid serialized action!"));
+        verify(destinations, never()).process(any(Action.class));
+    }
+
+    @Test
+    void testProcessingError() throws IOException {
+        Action action = buildValidAction();
+        String serializedAction = serializeAction(action);
+        when(destinations.process(eq(action))).thenReturn(
+                Uni.createFrom().failure(() -> new RuntimeException("I am a forced exception!"))
+        );
+        assertCountersIncrement(0, 1,
+                () -> inMemoryConnector.source("ingress").send(serializedAction));
+        verify(destinations, times(1)).process(eq(action));
+    }
+
+    private static Action buildValidAction() {
+        Action action = new Action();
+        action.setBundle("my-bundle");
+        action.setApplication("Policies");
+        action.setEventType("Any");
+        action.setTimestamp(LocalDateTime.now());
+        action.setAccountId("testTenant");
+        action.setPayload(Map.of("k", "v", "k2", "v2", "k3", "v"));
+        return action;
+    }
+
+    private void assertCountersIncrement(int expectedRejectedIncrement, int expectedProcessingErrorIncrement, Runnable runnable) {
+        double initialRejectedCount = rejectedCounter.count();
+        double initialProcessingErrorCount = processingErrorCounter.count();
+        runnable.run();
+        assertEquals(expectedRejectedIncrement, rejectedCounter.count() - initialRejectedCount);
+        assertEquals(expectedProcessingErrorIncrement, processingErrorCounter.count() - initialProcessingErrorCount);
+    }
```

Comment:
I don't know how to reset counters between tests (tried a few things which didn't work). If you have a better way to handle this, I'm open to suggestions :)

Reply:
You could try to get the metrics registry and clear it. It may even work by just injecting the registry.

Reply:
That's one of the thing I tried but it didn't work (or I couldn't find the good way maybe).

Reply:
Anyway this can be improved later. The current way to assert counters increment is working as intended, I just don't really like how it's done :)